lib/Model/Paste.php

```diff
@@ -164,7 +163,17 @@ public function getComment($parentId, $commentId = '')
      */
     public function getComments()
     {
-        return $this->_store->readComments($this->getId());
+        if ($this->_conf->getKey('discussiondatedisplay')) {
+            return $this->_store->readComments($this->getId());
+        }
+        return array_map(function($comment) {
+            foreach (array('created', 'postdate') as $key) {
```

Comment:
Do we have some test ensuring this new API feature? I.e. that the keys are properly deleted? (As that is really a thing one does not manually test/is cumbersome to test manually.)

Reply:
nope, I tested manually

Reply:
Maybe it would be a good idea to add that … was my saying… automated tests are always better.

Reply:
Oki, I'll add a test for that code path, also in the interest of retaining a high level of code coverage.

Reply:
Test added in a6058705a410b56fcc621e1fbfcbefe051d42881

Reply:
We did actually have tests that validated that the paste creation (and postdate for v1 pastes) is present, which I had to change to detect that the field is now _not_ present. I changed those in the current location, which is in the ControllerTest, and therefore also executed in the inherited ControllerDatabaseTest and ...GcsTest variants: https://github.com/PrivateBin/PrivateBin/pull/1295/files#diff-8977b8e7d85969361448323730a50efb0c879b761a480352b86ad95c7ef932baL715

So we do now have tests that ensure that:
- paste creation is not present (for the file, database & Gcs backends)
- comment creation is normally present, but absent when the configuration has it set to false
- database upgrade will work error free (for sqlite version with and without drop column support)
- the new table schema works and pastes can still be read from it